code/script.js

```diff
@@ -254,72 +271,165 @@ const checkQuestion = () => {
 
   // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
   // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
   if (category === 'hair' || category === 'eyes') {
 
-  } else if (category === 'accessories' || category === 'other') {
+    if (value === secret[category]) {
+      // Invoke filterCharacters
+      filterCharacters(true)
+      
+    }
+    else {
+      filterCharacters(false)
+      
+    }
+  }
+
```

Comment:
Something that might make the code look a bit "cleaner" is too have less line brakes when not needed. But it might be question about  preference.